validate_config.rb

```diff
@@ -4,13 +4,32 @@
 
 require 'rubocop'
 
-$stderr = StringIO.new
+def redirect_output
+  redirected_stdout = StringIO.new
+  redirected_stderr = StringIO.new
 
-# This will validate the config
-RuboCop::ConfigLoader.load_file('rubocop.yml')
+  default_stdout = $stdout
+  default_stderr = $stderr
 
-# Print out the errors to console
-puts $stderr.string
+  $stdout = redirected_stdout
+  $stderr = redirected_stderr
 
-# Exit with non-zero code so CI will mark as failed
-exit(1) unless $stderr.string.empty?
+  yield
+
+  $stdout = default_stdout
+  $stderr = default_stderr
+
+  [redirected_stdout, redirected_stderr]
+end
```

Comment:
You can probably shorten this by assuming that `$stdout == STDOUT`
```ruby
[StringIO.new, StringIO.new].tap do |out, err|
	$stdout, $stderr = out, err

	yield

	$stdout, $stderr = STDOUT, STDERR
end
```

Reply:
Ah, I didn't know about those constants. Will do.